src/components/player/Player.vue.js

```diff
@@ -0,0 +1,91 @@
+export default {
+  name: 'player',
+  props: {
+    data: {}
+  },
+  data() {
+    return {
+      id: {
+        'Progress': '.progress-bar',
+        'Duration': '.max-time',
+        'Pause':    '.button-pause',
+        'Play':     '.button-play',
+        'Audio':    'audio'
+      },
+      el: {},
+    }
+  },
+  methods: {
+    _pop(object) {
+      let key = Object.keys(object).pop()
+      return {
+        key: key,
+        value: object[key],
+        remainder: (delete object[key], object)
+      }
+    },
+    _controls(accumulator, container, object) {
+      let item = this._pop(object)
+      accumulator[item.key] = container.querySelector(item.value)
+      return Object.keys(object).length
+        ? this._controls.apply(null, [accumulator, container, item.remainder])
```

Comment:
Why apply here and not just `this._controls(accumulator, container, item.remainder)`?